### DISCUSSION TITLE
Truthiness of Option(False) should be False

### DISCUSSION THREAD

**Discussion by khaume** [fastapi/typer#597] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer

def main(some_flag: bool = typer.Option(False, help="help string")):
    if some_flag:
        print("some_flag was True/Truthy")

if __name__ == "__main__":
    run(main)
```

### Description

If I run this script from as CLI, then it works as expected, i.e. not providing the `--some-flag`, the code will not enter the `if` statement.

However, if I call the function from another python function, then `some_flag` is now an instance of the `Option` class, and `bool(some_flag)` is `True`, which I think is counter-intuitive. 

I am aware that I can check for the default value with `some_flag.default` which would then be `False`, but I think it would be great if there could be an implementation of the `__bool__` method in the `Option` class, which returns `bool(self.default)` for these cases.

### Operating System

Windows

### Typer Version

0.6.1

### Python Version

3.8.2

**Comment by tiangolo** [comment, marked as answer]
Hey there! Yep, that's because Typer only allowed setting the values as the function parameter "default", in that case, for it to work you would have had to pass the parameters manually when calling the function.

But now there's support for Annotated, to solve exactly this problem :tada: 

```Python
import typer
from typing_extensions import Annotated

def main(some_flag: Annotated[bool, typer.Option(help="help string")] = False):
    if some_flag:
        print("some_flag was True/Truthy")

if __name__ == "__main__":
    run(main)
    main()
```

It's available in Typer 0.9.0, just released :rocket:
